### ISSUE TITLE
Is there any though to VB supporting comments  "inline"

### ISSUE THREAD

**Issue by paul1956** [dotnet/vblang#271]
Not sure what the correct term is but similar to C# /* Comment */ as below

```
If x = 1 /* not zero */ then
    Dim y = 7 /* Special Flag */, z = 4 /* Version */
    ' Do Something
End If
```

I don't care about using /**/ it was just a familiar syntax, I don't know if this is possible without a very breaking change but it comes up a lot in trying to document flags. It would be really useful with new flag feature where it might not be a breaking change.

**Comment by ocdtrekkie** [comment]
I'm not sure I see the desperate need here, in your example, it's easy to put:

`Dim y = 7, z = 4 'y is a special flag, z is the version`

The biggest problem is if you used the format 'comment', it would break code for people who have apostrophes in their comments. In fact, the only thing you might be able to do non-breaking to change this, is that REM statements must be followed by a space, so you could do Rem(comments in here) and keep looking for code after it?

**Comment by AdamSpeight2008** [comment]
You could do
```vbnet
If x = 1 then
    Dim y = 7 ' Special Flag 
    Dim z = 4 ' Version
    ' Do Something
End If
```
or (in my opinion) rename the variable, leaving the comment to explain why you are doing something rather what you are doing. 
```vbnet
If x = 1 then
  Dim Special_Flag = 7
  Dim Version = 4 
  ' Do Something
End If
```

**Comment by tverweij** [comment]
Inline would be handy in some cases, but can indeed be done by splitting the line.
But Multi-line comments would be a nice thing to have.

**Comment by paul1956** [comment]
Yes, in the above cases you could split the statement into multiple lines but it you look through Roslyn source you will see 100's maybe 1,000's of examples of comments in the middle of lines and they are found all over the place, in parameter lists, in object creation... and I am not finding a single solution to replace all of them easily to replicate the coding style. In the case of arguments there is no name that explains what is being done that is why the comments. Some of the lines are very long and splitting them would make the function very long vs just a few lines.

Sometimes the comment is at the end of the line before a ). Splitting it would leave a line with just the ) and it would look odd. 

Today I just move everything to the end of the line but context gets lost.

.I actually like Rem(Comment Here)
```
Dim VersionString() as String = {"4"  Rem(Version), "0",..."7" Rem(Special Flag),..."8" Rem(Required)...}
```

**Comment by AdamSpeight2008** [comment]
@paul1956 Is this covered by #41 ?

**Comment by pricerc** [comment]
I'm of the (old) school of thought that if your line of code is more than ~80 characters, or if it can't fit onto a portrait-mode printout, then it is probably too long and should be split. In which case #41 and #65 will definitely help.

And if the Roslyn code (Which I haven't really looked at) has dubious coding practices such as no names to explain what is being done, I'm not sure it's a coding style I'd be recommending to anyone ;-)

I'm also curious to understand better the aversion some people have to 'tall' vs. 'wide' methods/method calls?

**Comment by paul1956** [comment]
@AdamSpeight2008  Yes #41 would help a lot, #65 is not something I do anymore after learning it didn't solve the comment problem.

@pricerc With wide screens and multiple monitors 80 character is no longer a practical limit. Where this is most frustrating is where binary is defined a byte at a time and contains 100's or more characters most of which are 0 and the comments make the few nonzero values stand out.

Since this is covered mostly by other issues this can be closes unless there is support for Rem(Comment) style inline comments.

**Comment by bandleader** [comment]
Also was hit by this recently, but I do also believe that just fixing #41 (same as #183 I think) would solve much of the problem.

If this were to be done, I personally don't like the `Rem(Comment here)` syntax at all, because keywords are noise in the code where they have no significance -- that's why we abandoned `REM` comments in favor of `'` years ago. A character syntax is much cleaner. I think `/* comment here */` is just fine, unless some are put off by C-esque syntax (these days I do so much JS that I don't mind).

**Comment by jasonmalinowski** [comment]
> The VB IDE and any code rewriting tools might leverage the fact that comments can only appear in certain places to optimize or simplify things.

I don't think in Roslyn we take any _intentional_ dependency on this fact. As you say, we already have to deal with C# so it's not a big deal. I could imagine a number of features having bugs though where we just accidentally forgot to copy trivia around and would break until we fixed them up. So any change here has a non-trivial test cost, but I would imagine most bugs would be of the "oops dumb bug" form vs. creating some sort of deep philosophical problems.

**Comment by paul1956** [comment]
I have a few other places comments would be nice like between lines within a multi-line parameter or argument list. For single line comment I can put the comment after the , but not for multiline. This would be non-breaking. 
```
If ProcessDirectory(FolderName,
    ' this is a test
    ' of multi line comments
    LastFileNameWithPath,
    LanguageExtension,
    FilesProcessed) Then
```
Also before a Then keyword. @jasonmalinowski  Today the IDE will refactor your code if you try to put anything before the Then.
```
If True ' Always do this
Then
```
Becomes
```
If True Then ' Always do this
```
If you look at all the place C# allows comment and that people actually put comments VB is very far behind.

**Comment by tverweij** [comment]
Solved in project Mercury
